src/stringUtils.js

```diff
@@ -21,47 +21,36 @@ const toFullName = ({ firstName, lastName, name }) =>
     : (!isBlank(lastName) ? `${properCaseName(lastName)}, ` : '') +
       (!isBlank(firstName) ? properCaseName(firstName) : '')
 
-/**
- * Render a moment object or ISO 8601 formatted string in the standard ui date format.  This format is
- * dd/MM/yyyy.  If the date parameter is not present or cannot be understood the function returns '--'.
- * @param date An object which is a moment, an ISO 8601 date formatted string or anything else including null or undefined.
- */
-const renderDate = (date) => {
-  const screenFormat = 'DD/MM/YYYY'
-  const notPresentString = '--'
-
-  const theMoment = typeof date === 'string' ? moment(date, iso8601DateFormat) : date
```

Comment:
the problem here was that `theMoment` wasn't constrained to particular type, could be `any`

Reply:
Might have been a union of `moment.Moment | string`?

You can provide jsdoc type hints if you need to - but the resulting code seems nicer anyway...